[DataGrid] Make possible to memoize rows and cells (#7846)
mui/mui-x: docs/data/migration/migration-data-grid-v5/migration-data-grid-v5.md

- The `GridActionsCellProps['api']` property was removed. Use `useGridApiContext` hook instead to get `apiRef`.
- The `GridActionsCellProps['getValue']` property was removed. Use `params.row` instead.
- The `GridFooterCellProps['getValue']` property was removed. Use `params.row` instead.

> - The `cellFocus`, `cellTabIndex` and `editRowsState` props are not passed to the component used in the row slot.
>   You can use the new `focusedCell` and `tabbableCell` props instead.
>   For the editing state, use the API methods.

Reviewer: ```suggestion
- The `cellFocus`, `cellTabIndex` and `editRowsState` props are not passed to the `Row` slot anymore.
  Use the `focusedCell` and `tabbableCell` props instead.
  For the editing state, use the API methods.
```
